struct.c

```diff
@@ -349,6 +376,113 @@ setup_struct(VALUE nstr, VALUE members)
     return nstr;
 }
 
+static VALUE
+setup_data(VALUE nstr, VALUE members)
```

Comment:
Why is it called `nstr`? (sounds like "n strings"). It's the new data subclass, so I think `subclass` is a good name.